src/install/lockfile.zig

```diff
@@ -3853,3 +3699,1012 @@ pub fn resolve(this: *Lockfile, package_name: []const u8, version: Dependency.Ve
+
+                    // var new_string_start = string_start[@boolToInt(is_quoted)..][@boolToInt(is_quoted2)..];
+                    // C.memmove(new_string_start.ptr, string_start.ptr, dep_name_len);
+                    // at_place[0] = '@';
+                    // const dependency_ver = new_string_start[0 .. @ptrToInt(cur.ptr) - @ptrToInt(new_string_start.ptr)];
+
+                    cur = cur[@boolToInt(is_quoted2)..]; // skip last '"'
+
+                    if (!advanceOverSkipInsignificant(&cur, "\n" ++ " " ** 4)) break :parse_sub_dependencies;
+                } // : parse_sub_dependencies
+
+                if (!advanceOverComptime(&cur, "\n" ++ " " ** 2)) continue :parse_dependency;
+            }
+
+            return error.@"Missing or invalid dependency field after indentation in yarn.lock";
+        }
+    }
+
+    /// Takes in a file owned by the caller to parse as a yarn.lock file
+    pub fn loadFromFile(allocator: Allocator, file: std.fs.File) !DependencyOverrides {
+        var body_pool = Npm.Registry.BodyPool.get(allocator);
+        defer Npm.Registry.BodyPool.release(body_pool);
+
+        // We over-allocate enough to alleviate the need for bounds checking.
+        try body_pool.data.readFile(file, over_allocated_space);
+        const afterFileContents = body_pool.data.list.allocatedSlice()[body_pool.data.list.items.len..];
+        for (afterFileContents[0..SENTINELS.len], SENTINELS) |*cur, sentinel| cur.* = sentinel;
+        return parse(allocator, body_pool.data.list);
+    }
+
+    inline fn parseMetaHash(meta_hash: *MetaHash, cur: *[]u8) !void {
+        comptime std.debug.assert(std.math.isPowerOfTwo(META_HASH_CHUNK_LEN));
+        var meta: []u8 = meta_hash.*[0..];
+        if (Environment.allow_assert) std.debug.assert(std.mem.allEqual(u8, meta, 0));
+        var bytes_left: u8 = META_HASH_LEN;
+
+        while (true) {
+            // 8 bytes represented across 16 characters means we should iterate 16 times
+            var chars_to_match: u8 = META_HASH_CHUNK_LEN * 2; // I hate giving types to values like this.
+            var shift: std.math.Log2Int(u8) = 4; // This will be used to shift by a half byte or 0 each time
+
+            while (chars_to_match > 0) : (chars_to_match -= 1) {
+                const c = cur.*[0];
+                meta[0] |= ((c + switch (c) {
+                    '0'...'9' => @as(u8, 0),
+                    'A'...'F', 'a'...'f' => @as(u8, 9),
+                    else => return error.@"Invalid meta_hash",
+                }) & 0xF) << shift;
+                cur.* = cur.*[1..];
+                shift ^= 4; // toggle between shifting by half a byte or not
+                meta = meta[shift >> 2 ..];
+            }
+
+            bytes_left -= META_HASH_CHUNK_LEN;
+            if (bytes_left == 0) break;
+            if (cur.*[0] != '-') return error.@"Invalid meta_hash";
+            cur.* = cur.*[1..];
+        }
+    }
+
```

Comment:
This function isn't actually used at the moment. Maybe it should be moved elsewhere or deleted?